### PR TITLE
test: Validate Service IP

### PR DESCRIPTION
**Pull request by jrajahalme**
Validate resolved service IP when waiting for service to become
available.

This will help avoid test flakes where kube-dns is returning a stale
IP of a service that was just removed, e.g., when running test with
`--force-deploy`.

Fixes: https://github.com/cilium/cilium/issues/16867
Signed-off-by: Jarno Rajahalme <jarno@isovalent.com>

### CODE REVIEW

**Comment by jrajahalme** [comment]
Multi-cluster test fails due to expecting DNS to return the ClusterIP of the local cluster, presumably. Artifact collection is broken, so can't confirm:
```
2021-07-14T12:08:47.498637237Z 🐛 Error waiting for service cilium-test/echo-other-node: Service IP "10.100.4.13" not found in nslookup output "Server:\t\t10.112.0.10\nAddress:\t10.112.0.10:53\n\nNon-authoritative answer:\nName:\techo-other-node.cilium-test.svc.cluster.local\nAddress: 10.112.10.71\n\n** server can't find echo-other-node.svc.cluster.local: NXDOMAIN\n\n** server can't find echo-other-node.cluster.local: NXDOMAIN\n\n** server can't find echo-other-node.c.***.internal: NXDOMAIN\n\n** server can't find echo-other-node.google.internal: NXDOMAIN\n\n** server can't find echo-other-node.svc.cluster.local: NXDOMAIN\n\n** server can't find echo-other-node.cluster.local: NXDOMAIN\n\n** server can't find echo-other-node.c.***.internal: NXDOMAIN\n\n** server can't find echo-other-node.google.internal: NXDOMAIN\n\n": 
```

File: connectivity/check/deployment.go
Comment (tklauser) [inline connectivity/check/deployment.go:650]: Could we maybe check for the full `Address: <IP>` line here? In its current state the check would mistakenly succeed e.g. for IP `192.168.1.1` if the address in `nslookup` output was in fact `192.168.1.10`.

(I think @pchaigno recently fixed a test flake in `cilium/cilium` caused by a similar piece of code)

Comment (jrajahalme) [reply connectivity/check/deployment.go]: Good point, done :-)

**Review by tklauser — APPROVED** [review]

**Comment by jrajahalme** [comment]
re-running AWS CI jobs due to infra problems (check for AWS ds scheduled jobs == 0 failed).

**Comment by tklauser** [comment]
> re-running AWS CI jobs due to infra problems (check for AWS ds scheduled jobs == 0 failed).

The failure is not due to infra problems, but an oversight when #400 was merged. The fix will be in #442, so I think this is good to merge once @michi-covalent approves.

**Review by michi-covalent — APPROVED** [review]

**Comment by jrajahalme** [comment]
rebased to resolve conflicts

**Comment by tklauser** [comment]
AKS failed on #367 which is a known flake: https://github.com/cilium/cilium-cli/pull/422/checks?check_run_id=3312947463

Merging.

**Pull request merged**
